var/spack/repos/builtin/packages/drishti/package.py

```diff
@@ -14,17 +14,24 @@ class Drishti(PythonPackage):
 
     homepage = "https://github.com/hpc-io/drishti-io"
     git = "https://github.com/hpc-io/drishti-io"
-    pypi = "drishti-io/drishti-io-0.4.tar.gz"
+    pypi = "drishti-io/drishti-io-0.6.tar.gz"
 
     maintainers("jeanbez", "sbyna")
 
     version("master", branch="master")
 
+    version("0.6", sha256="f1883e48de4a863f284f9c54febd2b04d04456848aea3e34c355310a3336e3b8")
+    version("0.5", sha256="08ae82dfa82872cde1f0a219e979b9c10d6c4a01f762066ea864a400cc144d78")
     version("0.4", sha256="bbbb272b4f6f44ae762f6cba28a2c589e15608691c559af0cc2f552590335d7b")
 
-    depends_on("darshan-util", type=("run"))
+    depends_on("darshan-util@3.4:", type=("run", "test"))
```

Comment:
Why is darshan-util required?

Reply:
pyDarshan requires the `libdarshan-util.so` library to be available in the path and with matching versions. When I try running `spack -v test run drishti` without darshan-util loaded or drishti loaded, the test seems to fail on my side. Should we explicitly load the dependencies before running a test, or does spack should load those implicitly? If the second case, I am not sure it is my setup, but that does not seem to work as expected as I get an error that the library was not in the path.

Reply:
In that case it should be a link dep. Can you add a comment explaining why it's required?

Reply:
So, I tried removing it and putting it back in, but still, for both cases, if I do not manually load darshan-util or drishti, the test fails with the following:

```
     29        raise RuntimeError('Could not find libdarshan-util.so! Is darshan-util installed? Please ensure one of the the following: 1) export LD_LIBRARY_PATH=<path-to-libdarshan-util.so>, or 2) darshan-parser can found using the PATH vari
           able, or 3) pkg-config can resolve pkg-config --path darshan-util, or 4) install a wheel that includes darshan-utils via pip.')
     30    RuntimeError: Could not find libdarshan-util.so! Is darshan-util installed? Please ensure one of the the following: 1) export LD_LIBRARY_PATH=<path-to-libdarshan-util.so>, or 2) darshan-parser can found using the PATH variable, or 3
           ) pkg-config can resolve pkg-config --path darshan-util, or 4) install a wheel that includes darshan-utils via pip.
```

What could be causing spack to not find the library unless I explicitly load it before the test?

Reply:
btw, even with link dep it does not seem to solve the problem

Reply:
What about `type=("link", "run")`?

Reply:
Even with that, apparently. When I try with:

```
spack uninstall --all drishti; spack install --test root drishti; spack test remove -y; spack -v test run drishti
```

Everything seems to work, but if I try with the following:

```
spack uninstall --all drishti; spack install drishti; spack test remove -y; spack -v test run drishti
```

The tests fail. I assume both should work since the first is also running the test during the installation. Am I missing something @adamjstewart or is there something else happening here?

Reply:
Hmm, that doesn't make any sense... Maybe @tldahlgren has some ideas?

Reply:
No. I wonder if there's something different in the build environment that we're not seeing in the run/test environment.

Trying to reproduce on my usual machine but had to rebase the PR to develop first before running into a failure installing py-matplotlib:

```
==> Installing py-matplotlib-3.7.2-5353k5rsjlzq4qqtvn4v7iooqo5gspxz [82/88]
==> No binary for py-matplotlib-3.7.2-5353k5rsjlzq4qqtvn4v7iooqo5gspxz found: installing from source
==> Error: NoLibrariesError: Unable to recursively locate libpng libraries in /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/libpng-1.2.57-ghafgcdwfhaiuryeee4d5ta4rej75jqs

/usr/WS1/dahlgren/releases/spack/var/spack/repos/builtin/packages/py-matplotlib/package.py:270, in setup_build_environment:
        267        for dep in self.spec.dependencies(deptype="link"):
        268            query = self.spec[dep.name]
        269            include.extend(query.headers.directories)
  >>    270            library.extend(query.libs.directories)
        271
        272        # Build uses a mix of Spack's compiler wrapper and the actual compiler,
        273        # so this is needed to get parts of the build working.
```

Trying again on another machine.

Reply:
What do the libs look like in that dir? That doesn't make sense either...

Reply:
> What do the libs look like in that dir? That doesn't make sense either...

```
opt/spack/linux-rhel8-broadwell/gcc-10.3.1/libpng-1.2.57-ghafgcdwfhaiuryeee4d5ta4rej75jqs/lib/:
libpng12.a
libpng12.so -> libpng12.so.0
 libpng12.so.0 -> libpng12.so.0.57.0
 libpng12.so.0.57.0
pkgconfig

opt/spack/linux-rhel8-broadwell/gcc-10.3.1/libpng-1.2.57-ghafgcdwfhaiuryeee4d5ta4rej75jqs/lib/pkgconfig:
libpng12.pc
libpng.pc
```

Reply:
@tldahlgren did you manage to get past that py-matplotlib error?

Reply:
Unfortunately, no.

Reply:
My setup seems to have installed py-matplotlib-3.4.3 instead of the 3.7.2 I see in your logs. I also tried rebasing with upstream and re-installing py-matplotlib (now 3.7.2) and I could not replicate the error.

Reply:
May need to add a `libs` property for ancient versions of libpng, for me the latest version has:
```console
> ls -lhA
total 2656
drwxr-sr-x  4 Adam  staff   128B Mar 31 14:39 libpng/
lrwxrwxr-x  1 Adam  staff    10B Mar 31 14:39 libpng.a@ -> libpng16.a
lrwxrwxr-x  1 Adam  staff    14B Mar 31 14:39 libpng.dylib@ -> libpng16.dylib
-rwxr-xr-x  1 Adam  staff   266K Mar 31 14:39 libpng16.16.39.0.dylib*
lrwxrwxr-x  1 Adam  staff    22B Mar 31 14:39 libpng16.16.dylib@ -> libpng16.16.39.0.dylib
-rw-r--r--  1 Adam  staff   1.0M Mar 31 14:39 libpng16.a
lrwxrwxr-x  1 Adam  staff    17B Mar 31 14:39 libpng16.dylib@ -> libpng16.16.dylib
drwxr-sr-x  4 Adam  staff   128B Mar 31 14:39 pkgconfig/
```

Reply:
@adamjstewart I'm not sure where this should be added, is it in our recipe?

@tldahlgren with that suggestion, did you managed to get it running?

Reply:
It would be in the `libpng` package.

Reply:
I am not sure I understood the changes required in `libpng`. Could you please advice?

Reply:
#39638